esp/services/esdl_svc_engine/esdl_store.hpp

```diff
@@ -0,0 +1,82 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _ESDL_STORE_HPP__
+#define _ESDL_STORE_HPP__
+
+#include "jlib.hpp"
+#include "jstring.hpp"
+#include "jptree.hpp"
+
+#ifdef ESDL_ENGINE_EXPORTS
+ #define esdl_engine_decl DECL_EXPORT
+#else
+ #define esdl_engine_decl
+#endif
+
+interface IEsdlStore : public IInterface
+{
+    virtual void fetchDefinition(const char* id, StringBuffer& esxdl) = 0;
+    virtual void fetchDefinition(const char* esdlServiceName, unsigned targetVersion, StringBuffer& esxdl) = 0;
+    virtual void fetchDefinitionByNameOnly(const char * name, StringBuffer & def) = 0;
+    virtual IPropertyTree* fetchBinding(const char* espProcess, const char* bindingName) = 0;
+    virtual bool isDefinitionBound(const char * esdldefid) = 0;
+    virtual bool isDefinitionBound(const char * esdldefname, int version) = 0;
+    virtual IPropertyTree* getDefinitionRegistry(const char * wsEclId, bool readonly) = 0;
+    virtual bool existsDefinition(const char * definitionid) = 0;
+    virtual bool existsDefinition(const char * servicename, unsigned ver) = 0;
+    virtual bool existsMethodDef(const char * esdlDefinitionName, unsigned ver, StringBuffer & esdlServiceName, const char * methodName) = 0;
+    virtual void addDefinition(IPropertyTree * queryRegistry, const char * name, IPropertyTree *definitionInfo, StringBuffer &newId, unsigned &newSeq, const char *userid, bool deleteprev) = 0;
+    virtual int publishMethod(const char * espProcName, const char * espBindingName, const char * srvDefId, const char * methodName, IPropertyTree * attributesTree, bool overwrite, StringBuffer & message) = 0;
+    virtual int publishBinding(const char * bindingName, IPropertyTree * methodsConfig, const char * espProcName, const char * espPort, const char * esdlDefinitionName,
+                               int esdlDefinitionVersion, const char * esdlServiceName, StringBuffer & message, bool overwrite, const char * user) = 0;
+    virtual IPropertyTree* getBindingTree(const char * espProcName, const char * espBindingName, StringBuffer & msg) = 0;
```

Comment:
This is another one that may evolve as you work on other issues.  We really need to support more than one esdl service being bound per port.  Also may evolve when not using DALI as back end.